lib/internal/modules/cjs/loader.js

```diff
@@ -753,7 +756,7 @@ Module.prototype._compile = function(content, filename) {
       undefined,
       false,
       undefined,
-      [],
```

Comment:
I share @devsnek ‘s concern here, it does not seem to worth a lot to deprecate this in CJS

Reply:
Note `process` and `Buffer` are not being deprecated in CJS, and never will, rather `global.process` and `global.Buffer` are only.

Reply:
If we are still stuck with the `node_modules` detection for `new Buffer` before emitting warnings, TBH this is just going to be even more disruptive than that...

Reply:
> If we are still stuck with the `node_modules` detection for `new Buffer` before emitting warnings

@joyeecheung can you share what you mean by this statement? I'm not sure I follow.

> I believe most users do not require Buffer or process before using them - not even the majority of our tests do that.

None of these usages have to change at all, including in our tests.

Reply:
To be very clear here - any access to `process` or `Buffer` in any existing Node.js script or library will never be deprecated or provide a deprecation warning, unless it is an access to the `global.process` or `global.Buffer` member forms explicitly.

Reply:
There’s still the refactoring hazard of converting CJS to ESM to worry about - that it’s not technical breakage doesn’t make it problem-free.

Reply:
@ljharb can you relate how the breakage in this PR differs from other breakage hazards of migrating to ESM / why this kind of movement towards a more uniform ESM environment would be detrimental?

Reply:
@bmeck there will be a list of things to account for when migrating; the more things are in that list, the harder it becomes. That includes any module-level-visible identifiers. I don’t want to add unnecessary things to that list.

Reply:
Sorry, didn’t notice the shadowing in the compileFunction wrapper - however I still suspect that there is a fair amount of access to these via global in the user land, especially for packages that detect Node environments, and emitting warnings in these situations put us back into the Buffer constructor problem - the prevalent usage of this makes it annoying enough for users and for package maintainers that we are stuck with detecting whether the invocation comes from node_modules before emitting the warnings.

Reply:
@ljharb that is true if we want to have legacy code remain unchanged during migration, but there is the counterpoint that moving to a new system requires changes already and that there could be benefits by these breaking changes. I personally find the fact that the code is being migrated to a new system (ESM) does not give the same backwards compat concerns as keeping code running without changes in the prior system.

Reply:
@joyeecheung I can aim to run some tests on common libraries here. Are there existing workflows for this you can recommend? Otherwise I can aim to do a test of the top 1000 npm packages for example.

Reply:
I ran some tests and `global.process` usage was still very prevalent including in eg lodash.

As a result I've updated with an approach that doesn't deprecate `global.process` in CommonJS anymore.

Reply:
does that mean you’ll have a property on the global object that’s not a global variable? while that’s technically possible, it’s pretty confusing, and still breaks `typeof process` checks.